backend/libbackend/api.ml

```diff
@@ -178,7 +179,14 @@ let functions ~username =
          ; return_type = Dval.tipe_to_string v.return_type
          ; preview_safety = v.preview_safety
          ; infix = List.mem ~equal:( = ) v.infix_names k
-         ; deprecated = v.deprecated })
+         ; deprecated = v.deprecated
+         ; is_supported_in_query =
```

Comment:
This function looks weird, and I'm not sure what it does. Two `List.exists` seems suspect.

I'd suggest make Sql_compiler.compilerSupportedFns a Set (`Tc.StrSet` would do I think) and then just lookup whether it's in there, instead of a list traversal.

Reply:
Updated to a set.  There is still the List.exists because fn.prefix_names is a list.  I didn't see a better key to use on fn, so I wrote this so if *any* prefix name is in the supported list, the fn is supported.  I haven't found an example of a fn with more than one prefix name yet though.

Can you point me at docs or something for StrSet?  I found tc.ml, which includes Tablecloth, but Tablecloth's docs don't match what seems to be available (based on error messages).  And the ocmal set docs I found didn't seem the same either, so I'm not sure what this thing actually is!  I really need to get merlin working in vscode, I feel like I'm flying blind a lot of the time and maybe that would help.

Reply:
Ah, that's helpful. Yes, there's only ever one prefix_names, afaik. We preprocess it out later so it's actually kinda useless.

You can see some uses of StrSet in client/src/analysis/Sync.ml.

Reply:
I also don't have merlin setup and it bothers me 😭